addons/account/models/account_move.py

```diff
@@ -2713,16 +2715,7 @@ def _check_constrains_account_id(self):
             if control_type_failed or control_account_failed:
                 raise UserError(_('You cannot use this general account in this journal, check the tab \'Entry Controls\' on the related journal.'))
 
-    @api.constrains('tax_ids', 'tax_line_id')
-    def _check_tax_lock_date1(self):
-        for line in self:
-            if line.date <= (line.company_id.tax_lock_date or date.min):
-                raise ValidationError(
-                    _("The operation is refused as it would impact an already issued tax statement. " +
-                      "Please change the journal entry date or the tax lock date set in the settings ({}) to proceed").format(
-                        line.company_id.tax_lock_date or date.min))
-
-    @api.constrains('credit', 'debit', 'date')
```

Comment:
@william-andre I checked with current dev and it still doesn't allow to add/remove taxes for invoice created on or before tax lock date. That's because on removing taxes from existing invoice lines, `line_ids` is updated, ie. unlinked tax related move lines but hasn't removed it physically, hence still considers those move lines. Can you recheck ?

Reply:
Indeed, 2 major changes happened since those constraints were added:
* the tax refactoring
* account-apocalypse

The tax refactoring should have added `tax_repartition_line_id`, `tax_exigible` and `tax_base_amount` in `_check_tax_lock_date1`, and the apocalypse made that same method useless. However those 3 fields should now be added in the check of `_check_tax_lock_date2`
It would also be good if those 3 fields were readonly by default.

Reply:
done

Reply:
Sorry, what I said in my last message is wrong. You still need `_check_tax_lock_date1`, and the check for the 3 fields cited above should be done there.
Here is why:
* You can never change `tax_exigible` and `tax_base_amount` after the lock date, no matter what.
* If you remove `tax_repartition_line_id` on the tax line, you can then remove `tax_ids` on the expense/income line because the 'onchange' will not trigger and `tax_line_id` will be removed on the tax line. You should never be able to change any of those fields, no matter what.

Reply:
@william-andre I kept `_check_tax_lock_date1` and did as you ask, can you check ?